File: lua/selectIslands.lua
-- Works with Rizom 2022.0

--- Get islands by bounding box size, islands with U or V <= maxU or maxV will be selected.
-- @param maxU (number) Maximum U size in pixels.
-- @param maxV (number) Maximum V size in pixels.
local function selectIslandsByPixelSize(maxU, maxV)
	ZomSet({ Path = "Vars.EditMode.ElementMode", Value = 3 })

	local currentUVMap = ZomGet("Lib.CurrentUVSetName")
	local resolution = ZomGet("Lib.UVSets." .. currentUVMap .. ".RootGroup.TopoStable.Pack.MapResolution")
	local islands = ZomGet("Lib.Mesh.Islands")

	local islandsToSelect = {}
	for id, _ in pairs(islands) do
		local uMinus, uPlus, vMinus, vPlus = table.unpack(ZomEval("Lib.Mesh.Islands." .. id .. ".GetBBoxUVW"))

		local u = (uPlus - uMinus) * resolution
		local v = (vPlus - vMinus) * resolution

		if u <= maxU or v <= maxV then
			table.insert(islandsToSelect, tonumber(id))
		end
	end

	if #islandsToSelect == 0 then
		print("No islands meet the criteria")
		return
	end

	ZomSelect({
		PrimType = "Island",
		WorkingSet = "Visible",
		ResetBefore = true,
		Select = true,
		IDs = islandsToSelect,
		List = true,
	})
end

selectIslandsByPixelSize(3, 3)

File: python/select_islands.py
"""Select islands by pixel size.

Tested on versions:
    - Rizom 2024.1
"""

from typing import Any

App: Any = None


def select_islands_by_pixel_size(max_u: int, max_v: int):
    """Select islands whose U or V pixel width is <= the given value.

    Args:
        max_u (int): Minimum U size to select.
        max_v (int): Minimum V size to select.

    """
    current_uvmap: str = App.Get("Lib.CurrentUVSetName")
    resolution: int = App.Get(f"Lib.UVSets.{current_uvmap}.RootGroup.Properties.Pack.MapResolution")
    islands: dict = App.Get("Lib.Mesh.Islands")

    islands_to_select = []
    for id in islands.keys():
        u_minus, u_plus, v_minus, v_plus, *_ = App.Eval(f"Lib.Mesh.Islands.{id}.GetBBoxUVW")
        u = (u_plus - u_minus) * resolution
        v = (v_plus - v_minus) * resolution

        if u <= max_u or v <= max_v:
            islands_to_select.append(int(id))

    App.Select(
        {
            "PrimType": "Island",
            "WorkingSet": "Visible",
            "ResetBefore": True,
            "Select": True,
            "IDs": islands_to_select,
            "List": True,
        }
    )
